### PR TITLE
Fix testCannotShrinkLeaderIndex

### PR DESCRIPTION
**Pull request by AthenaEryma**
This test should no longer pass when the functionality it is intended to
test is broken, as it now indexes a number of documents and verifies
that the index is staying on the same step until after indexing and
replication of those documents is finished. This prevents the test from
passing if the leader index progresses in its lifecycle during that time.

Relates to https://github.com/elastic/elasticsearch/issues/34648
To be merged after https://github.com/elastic/elasticsearch/pull/39088

---

This PR should fail CI until CCR integration of shard history retention leases is complete (see https://github.com/elastic/elasticsearch/issues/37165)

### CODE REVIEW

File: x-pack/plugin/ilm/qa/multi-cluster/src/test/java/org/elasticsearch/xpack/indexlifecycle/CCRIndexLifecycleIT.java [hunk 1]
```diff
@@ -374,6 +367,20 @@ public void testCannotShrinkLeaderIndex() throws Exception {
                     assertILMPolicy(client(), indexName, policyName, "hot", "unfollow", "wait-for-indexing-complete");
                 });
 
+                // Index a bunch of documents and wait for them to be replicated
+                for (int i = 0; i < 50; i++) {
+                    index(leaderClient, indexName, Integer.toString(i));
+                }
+                assertBusy(() -> {
+                    for (int i = 0; i < 50; i++) {
+                        index(client(), indexName, Integer.toString(i));
```
Comment (dakrone) [inline x-pack/plugin/ilm/qa/multi-cluster/src/test/java/org/elasticsearch/xpack/indexlifecycle/CCRIndexLifecycleIT.java]: I'm not sure I follow this, we used to index a document on the leader and then make sure it existed on the follower, but now we index 50 documents on the leader and then index 50 documents on the follower, do we need to change this to check for the documents' existence?

**Review by dakrone — COMMENTED** [review]
I left a question about the test

File: x-pack/plugin/ilm/qa/multi-cluster/src/test/java/org/elasticsearch/xpack/indexlifecycle/CCRIndexLifecycleIT.java (hunk 1 above)
Comment (gwbrown) [reply x-pack/plugin/ilm/qa/multi-cluster/src/test/java/org/elasticsearch/xpack/indexlifecycle/CCRIndexLifecycleIT.java]: You are absolutely correct. Fixed.

**Review by dakrone — APPROVED** [review]
LGTM when CI is happy

**Pull request merged**